lib/wrapeth/provider.ex

```diff
@@ -1,85 +1,60 @@
-        {:ok, accounts} = module_name.eth_accounts(url: node_url)
-        accounts
+      def eth_accounts(_args \\ []) do
+        call_client(:eth_accounts)
       end
 
       @impl true
-      def get_balance(address, block \\ "latest") do
-        {module_name, node_url} = get_module_and_url()
-        {:ok, balance} = module_name.eth_get_balance(address, block, url: node_url)
-        balance
+      def eth_get_balance(address, block \\ "latest", _args \\ []) do
+        call_client(:eth_get_balance, [address, block])
       end
 
       @impl true
-      def get_block_number() do
-        {module_name, node_url} = get_module_and_url()
-        {:ok, block_number} = module_name.eth_block_number(url: node_url)
-        block_number
+      def eth_block_number(_args \\ []) do
+        call_client(:eth_block_number)
       end
 
       @impl true
-      def get_block_by_number(number, full) do
-        {module_name, node_url} = get_module_and_url()
-        {:ok, block} = module_name.eth_get_block_by_number(number, full, url: node_url)
-        block
+      def eth_get_block_by_number(number, full, _args \\ []) do
+        call_client(:eth_get_block_by_number, [number, full])
       end
 
       @impl true
-      def get_transaction_count(address, block \\ "latest") do
-        {module_name, node_url} = get_module_and_url()
-        {:ok, tx_count} = module_name.eth_get_transaction_count(address, block, url: node_url)
-        tx_count
+      def eth_get_transaction_count(address, block \\ "latest", _args \\ []) do
+        call_client(:eth_get_transaction_count, [address, block])
       end
 
       @impl true
-      def get_block_transaction_count_by_number(block \\ "latest") do
-        {module_name, node_url} = get_module_and_url()
-
-        {:ok, tx_count} =
-          module_name.eth_get_block_transaction_count_by_number(block, url: node_url)
-
-        tx_count
+      def eth_get_block_transaction_count_by_number(block \\ "latest", _args \\ []) do
+        call_client(:eth_get_block_transaction_count_by_number, [block])
       end
 
       @impl true
-      def get_gas_price() do
-        {module_name, node_url} = get_module_and_url()
-
-        {:ok, gas} =
-          module_name.eth_gas_price(url: node_url)
-
-        gas
+      def eth_gas_price(_args \\ []) do
+        call_client(:eth_gas_price)
       end
 
       @impl true
-      def get_transaction_receipt(hash) do
-        {module_name, node_url} = get_module_and_url()
-
-        {:ok, receipt} =
-          module_name.eth_get_transaction_receipt(hash, url: node_url)
+      def eth_get_transaction_receipt(hash, _args \\ []) do
+        call_client(:eth_get_transaction_receipt, [hash])
+      end
 
-        receipt
+      defp call_client(method_name, args \\ [], _arg \\ []) do
+        case apply(@client_type, method_name, args ++ [[{:url, @node_url}]]) do
+          {:ok, value} -> {:ok, value}
+          {:error, error} -> {:error, error}
+        end
```

Comment:
Doesn't this seem weird?
You have 2 cases, and you're retrieving the same thing in both.

Reply:
I just thought that the code would be more readable like this, but, in fact, this case statement is trivial.